src/Module.php

```diff
@@ -152,6 +154,16 @@ function (RegisterTemplateRootsEvent $e) {
                     }
                 }
             );
+
+            Event::on(
+                UrlManager::class,
+                UrlManager::EVENT_REGISTER_SITE_URL_RULES,
+                function (RegisterUrlRulesEvent $event) {
+                    $partsKitDir = self::$config['partsKit']['directory'];
+
+                    $event->rules[$partsKitDir] = ['template' => 'viget-base/parts-kit/index'];
```

Comment:
I really wanted this to point to a controller, but I could not get it to hit the controller action. I think because it's a submodule, there's something wacky about the controller map.

Reply:
Unrelated, but your comment got me thinking... it would be rad to somehow initialize the base module without needing to add it as a sub module. 

https://github.com/nystudio107/craft-autocomplete#bootstrapping-yii2-extension

Somehow their package auto-bootstraps itself. Thought it might be cool, but could be too much of a breaking change.

Reply:
🤯 andrew welch what a wizard

Reply:
Ooooh interesting. Looks like you can implement the <a href="https://www.yiiframework.com/doc/api/2.0/yii-base-bootstrapinterface">BootstrapInterface</a> and get auto-boostrapped (instead of needing to add to the bootstrap in app.php). Will investigate further

Reply:
After a bunch of investigation, I did finally get this to auto bootstrap.

1. Had to implement the `BootstrapInterface`

```php
class Module extends \yii\base\Module implements \yii\base\BootstrapInterface
```

2. Then moved everything from the `init()` into `boostrap()`

3. It looks like you have to manually define a controller map because of when auto bootstrapped modules are initialized

```php
use viget\base\controllers\PartsKitController;

Craft::$app->controllerMap['parts-kit'] = PartsKitController::class;
```

And the one gotcha that I haven't been able to figure out is that functional tests don't work anymore when the module is auto bootstrapped.

Reply:
I'm an idiot. I don't know why I initialized the module as a submodule vs bootstrapping on it's own.

https://github.com/vigetlabs/craft-viget-base/pull/43/files#diff-8b042e3f94ca5c59a7cd990b950aec0073ea84fe811e7b22be51158d7b180d56R15